### PR TITLE
DYN-5753 Add workflow to check file size

### PR DESCRIPTION
**Pull request by avidit**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Add workflow to check file size.

- Github allows committing file with size greater than 50 MB but files larger than 50 MB blocks waiter sync job which syncs this repo with the internal mirror. e.x [PSETDEVREL-1043](https://jira.autodesk.com/browse/PSETDEVREL-1043)

- This workflow will check file sizes for all changed files in a PR and fails the build if any files are larger than 50 MB.

[About large files on GitHub](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github#file-size-limits)
### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: .github/workflows/check_file_size.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Check file size
+on: push
```
Comment (QilongTang) [inline .github/workflows/check_file_size.yml:2]: I think this means per commit push to whatever branch has a PR open?

Comment (avidit) [reply .github/workflows/check_file_size.yml]: Updated to run on pull request

File: .github/workflows/check_file_size.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: Check file size
+on: pull_request
```
Comment (sm6srw) [inline .github/workflows/check_file_size.yml:2]: Should be also do this on every push?

**Comment by sm6srw** [comment]
I assume that this will fail the PR if we mark the action as required?

**Comment by mjkkirschner** [comment]
@avidit @QilongTang @sm6srw can we merge this in?

File: .github/workflows/check_file_size.yml (hunk 2 above)
Comment (avidit) [reply .github/workflows/check_file_size.yml:2]: Initially I added it to run it on `push`. We can change it to that if on push is better option.

**Comment by avidit** [comment]
> @avidit @QilongTang @sm6srw can we merge this in?

I think we can merge this.

**Pull request merged**